src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java

```diff
@@ -459,6 +461,42 @@ public void testImportControlFileName2() throws Exception {
                 getPath("InputImportControlTestRegexpInFile2.java"), expected);
     }
 
+    @Test
+    public void testPathWithNonAsciiCharacters() throws Exception {
+        final Path dirWithNonAsciiCharacters = temporaryFolder.toPath()
+                .resolve("\\u00e6\\u00f8\\u00e5")
+                .resolve("com/puppycrawl/tools/checkstyle/checks/imports/importcontrol");
+        dirWithNonAsciiCharacters.toFile().mkdirs();
+        final Path configInDirWithNonAsciiCharacters = dirWithNonAsciiCharacters.resolve(
+                "InputImportControl8.java");
+
+        for (String file : List.of("InputImportControl8.java", "InputImportControlBroken.xml")) {
+
+            Files.copy(Paths.get(getPath(file)), dirWithNonAsciiCharacters.resolve(file));
+        }
+
+        try {
+            final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+            verifyWithInlineConfigParser(configInDirWithNonAsciiCharacters.toString(), expected);
+            assertWithMessage("Test should fail if exception was not thrown").fail();
```

Comment:
Hmm, checkstyle should work well according to issue, please explain why you expect exception